sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala

```diff
@@ -189,18 +191,19 @@ private[ui] class SparkPlanGraphCluster(
     name: String,
     desc: String,
     val nodes: mutable.ArrayBuffer[SparkPlanGraphNode],
-    metrics: Seq[SQLPlanMetric])
-  extends SparkPlanGraphNode(id, name, desc, metrics) {
+    metrics: Seq[SQLPlanMetric],
+    codegenStageId: Option[Int] = None)
```

Comment:
can we include the codegen id when building the `name`? Then we don't need to change the class signature here.